metadata-jobs/mce-consumer/src/main/java/com/linkedin/metadata/kafka/MetadataChangeProposalsProcessor.java

```diff
@@ -100,29 +100,50 @@ public void consume(final ConsumerRecord<String, GenericRecord> consumerRecord)
         log.debug("Record {}", record);
       }
 
-      MetadataChangeProposal event = new MetadataChangeProposal();
+      final MetadataChangeProposal event;
       try {
         event = EventUtils.avroToPegasusMCP(record);
 
-        Urn entityUrn = event.getEntityUrn();
-        String aspectName = event.hasAspectName() ? event.getAspectName() : null;
-        String entityType = event.hasEntityType() ? event.getEntityType() : null;
-        ChangeType changeType = event.hasChangeType() ? event.getChangeType() : null;
-        MDC.put(MDC_ENTITY_URN, Optional.ofNullable(entityUrn).map(Urn::toString).orElse(""));
-        MDC.put(MDC_ASPECT_NAME, aspectName);
-        MDC.put(MDC_ENTITY_TYPE, entityType);
-        MDC.put(
-            MDC_CHANGE_TYPE, Optional.ofNullable(changeType).map(ChangeType::toString).orElse(""));
-
-        if (log.isDebugEnabled()) {
-          log.debug("MetadataChangeProposal {}", event);
-        }
-        String urn = entityClient.ingestProposal(systemOperationContext, event, false);
-        log.info("Successfully processed MCP event urn: {}", urn);
-      } catch (Throwable throwable) {
-        log.error("MCP Processor Error", throwable);
-        log.error("Message: {}", record);
-        KafkaListenerUtil.sendFailedMCP(event, throwable, fmcpTopicName, kafkaProducer);
+        systemOperationContext.withQueueSpan(
+            "consume",
+            event.getSystemMetadata(),
+            consumerRecord.topic(),
+            () -> {
+              try {
+                Urn entityUrn = event.getEntityUrn();
+                String aspectName = event.hasAspectName() ? event.getAspectName() : null;
+                String entityType = event.hasEntityType() ? event.getEntityType() : null;
+                ChangeType changeType = event.hasChangeType() ? event.getChangeType() : null;
+                MDC.put(
```

Comment:
These do make our logs longer and might not be as useful with the trace-ids, not sure if worth removing or not.

Reply:
I'd consider removing them if/when the traces are tracking this information or if we decide the logging is too much. We can always leave the context here and reconfigure logging to remove them as well.